soci/ztoc_test.go

```diff
@@ -607,6 +609,28 @@ func TestWriteZtoc(t *testing.T) {
 	}
 }
 
+func TestReadZtocInWrongFormat(t *testing.T) {
+	testCases := []struct {
+		name           string
+		serializedZtoc []byte
+	}{
+		{
+			name:           "ztoc unmarshal returns error and does not panic",
+			serializedZtoc: genRandomByteData(50000),
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			zs := ZtocSerializer{}
+			r := bytes.NewReader(tc.serializedZtoc)
+			if _, err := zs.Unmarshal(r); err == nil {
+				t.Fatalf("")
```

Comment:
Can we be more descriptive here? 

"Expected error but got nil"

Reply:
Good catch. Done.